client/assets/stylesheets/shared/mixins/_mixins.scss

```diff
@@ -1,8 +1,8 @@
 
 @import 'banner';
-@import 'breakpoints';
+@import '~@automattic/components/src/styles/_breakpoints.scss';
 @import 'calc';
-@import 'clear-fix';
+@import '~@automattic/components/src/styles/_clear-fix.scss';
```

Comment:
I'd like to know whether you have longer term thoughts on this change? It seems that we have moved some components to the package arbitrarily (we moved them as they were needed and when they were considered "leaf" components), and now we are moving some arbitrary bits of SASS as well.

Would we expect all of the "framework"-type SASS to move to the components package?

---

I don't really like importing from an arbitrary `src` path, that seems a bit like accessing private package internals. I wouldn't want e.g. the components to be used from `src` 😕 

If we have SASS that is public in this package, I'd suggest it go in a directory that makes that clear. I also wonder if there would be benefit from categorization under this directory like we have in Calypso, but I don't feel strongly about that:

```sass
// "Public" directory
@import '~@automattic/components/sass/_breakpoints.scss';
// Categorized
@import '~@automattic/components/sass/mixins/_breakpoints.scss';
```

Reply:
> I'd like to know whether you have longer term thoughts on this change? 

In general my long term hope is that the use of these utilities outside the package itself is greatly reduced as we offer more granular components that can be composed therefore, reducing the need for additional styling and these utilities. 

That way we might be able to stop exporting some of the utilities.

The remaining utilities could be grouped and exposed in their own packages e.g.

```scss
@import '~@a8c/breakpoint`;

@debug $a8c-breakpoints;

@include a8c-breakpoint('>large') {
  // ...
}
```

> Would we expect all of the "framework"-type SASS to move to the components package?

There's a good chance that it might all move depending on what parts are deemed reusable! 

I'm keen to incrementally pull across and refactor what we need rather than moving everything across up front and exposing it - that makes it harder for Calypso to iterate, adds more code to be maintained in `@automattic/components` and increases the likelyhood of what is exposed requiring breaking changes.

I'm keen to only move what's necessary across leaving Calypso free to change and iterate without impacting others.

> If we have SASS that is public in this package, I'd suggest it go in a directory that makes that clear. 

I like it 👍 

> I also wonder if there would be benefit from categorization under this directory like we have in Calypso, but I don't feel strongly about that

My personal preference is definitely to group by concern (e.g. typography, breakpoints) rather than type (e.g. variable, mixin, function) because concerns are usually coupled and it saves me looking across multiple files and forming a mental picture of how code across multiple files fits together... but if there's a prior pattern I'm OK with towing the line!

Reply:
I actually like this idea of only moving the modules that are needed for now. Similar to what I said about the changes in `_typogrpahy.scss`, it could be that we never end up sharing those files with anything outside calpyso.
I think we can develop stronger ideas about what belongs in and out of `@automattic/components` once we've been using it for longer.

However I do agree that since this is now part of the package's public API that the files shouldn't be getting imported from `src`.

Reply:
> it could be that we never end up sharing those files with anything outside calpyso.

I could see us use shared Typography sass to make things consistent across Calypso and WP Admin for Atomic sites.

Reply:
> I could see us use shared Typography sass to make things consistent across Calypso and WP Admin for Atomic sites.

That's true and would be pretty awesome. By the time we are ready for that we could be including it from somewhere like `@8c/typography`. Or maybe it'll be a different solution. I'm just biased towards deciding that stuff as late as possible.

Reply:
BTW a `__experimentalText` component just merged yesterday. Unfortunately the `text()` mixin  (CSS-in-JS)isn't exposed yet, but the component should be able to support a bunch of use cases by itself.

https://github.com/WordPress/gutenberg/pull/18495